SDWebImage/Core/SDImageIOAnimatedCoder.m

```diff
@@ -152,11 +186,17 @@ + (NSUInteger)imageLoopCountWithSource:(CGImageSourceRef)source {
 }
 
 + (NSTimeInterval)frameDurationAtIndex:(NSUInteger)index source:(CGImageSourceRef)source {
+    NSTimeInterval frameDuration = 0.1;
+    
+    // Earily return when application will be terminated.
+    if (self.willTerminated) {
```

Comment:
Don't add guard on this method. Wrong design.

This is a helper method, add to the caller.

Reply:
Done.